### DISCUSSION TITLE
How to customize ui.range() labels?

### DISCUSSION THREAD

**Discussion by MohAid** [zauberzeug/nicegui#3433] [Q&A] [answered]
### Question

Hi,
I am trying to customize the left and right labels of the `ui.range()` element but I failed, any help please?

[Quasar Example](https://quasar.dev/vue-components/range/#example--custom-label-values)

And this is one of my attempts to get it:
```py
ui.range(min=0, max=10000, step=10, value={'min': 2500, 'max': 7500}).classes('w-full')\
    .props('v-model="model"')\
    .props("""left-label-value=model.min+" USD" """)\
    .props("""right-label-value=model.max+" USD" """)\
    .props('label-always')
```

And an extra need, I want to present the number from `model.min` or `model.max` at the labels in a format such 1,000.00 USD.

**Comment by falkoschindler** [comment, marked as answer]
See #3426 for a very similar problem. A possible solution is to update the labels from an event handler:
```py
ui.range(min=0, max=100, step=1, value={'min': 0, 'max': 100}).props('label-always') \
    .on_value_change(lambda e: e.sender.props(f'left-label-value={e.value["min"]}px'))
```

**Reply by MohAid** [reply]
Yes @falkoschindler that works fine, and thank you for your quick answer.
(I am sorry I only searched at the discussions)

**Reply by falkoschindler** [reply]
No worries, such issues and discussions can be hard to find.
Your question just appeared very familiar to me, so I searched extra carefully. 🕵🏻‍♂️

**Reply by MohAid** [reply]
I liked the solution, it is a technique for solving many other cases, I wonder how I didn't think that way!
You give great advice
